templates/site/js/catalog-search.js

```diff
@@ -0,0 +1,162 @@
+
+
+    if(inputBox.value !="") {
+        inputBox.setAttribute("aria-expanded", "true");
+
+        if (fuse !== null) {
+            queryablesArray = fuse.search(inputBox.value);
+
+            if(queryablesArray.length > 0){
+                queryablesArray.forEach((queryableItem, queryableKey) => {
+                    if(queryableItem.matches.length > 0)
+                    {
+                        queryableItem.matches.forEach((matchItem, matchKey) => {
+                            var queryResultListItem = document.createElement("li");
+                            var listItemFont = document.createElement("h5");
+                            listItemFont.classList.add("margin-unset");
+
+                            queryableResultButton = document.createElement('a');
+                            queryableResultButton.innerText = matchItem.value;
+                            queryableResultButton.setAttribute('role', 'button');
+                            queryableResultButton.setAttribute('queryablekeytype', matchItem.key);
+                            queryableResultButton.setAttribute('queryablekeyvalue', queryableItem.item.key);
+                            queryableResultButton.id = "match" + matchKey;
+
+                            queryableResultButton.addEventListener('click', function (event) {
+                                selectSearchResults(inputBox, event.target.id);
+                            })
+
+                            listItemFont.appendChild(queryableResultButton);
+                            queryResultListItem.appendChild(listItemFont);
+                            queryResultList.appendChild(queryResultListItem);
+                        });
+                    }
+                })
+            }
+            else{
+                inputBox.setAttribute("aria-expanded", "false");
+            }
+        }
+    }
+    else{
+        inputBox.setAttribute("aria-expanded", "false");
+    }
+}
+
+function selectSearchResults(inputBox, buttonID){
+    var listButton = document.getElementById(buttonID);
+    inputBox.value = listButton.innerText;
+    inputBox.setAttribute("aria-expanded", "false");
+    inputBox.classList.add("returned-result")
+}
+
+function removeReturnedResultStyle(inputBox){
+    inputBox.classList.remove("returned-result");
+}
+
+function addDefaultSearchAttributes(inputBox){
+    inputBox.setAttribute("disabled", "disabled");
+}
+
+function removeDefaultSearchAttributes(inputBox){
+    inputBox.removeAttribute("disabled");
+}
+
+//TODO get the focus onto the first list item when search input box detects arrowdown
+function focusOnResults(){
+
+    const resultList = document.getElementById("suggestedWordOutputList");
+
+    var resultLinkArray = resultList.querySelectorAll('li h5 a')
+    //if(keyPressed === "ArrowDown"){
+        var firstLink = document.getElementById(resultLinkArray[0].id)
+        //console.log(firstLink);
```

Comment:
Do I really have to comment on every single one? I don't want to have to write a comment like this https://github.com/DACCS-Climate/stac-search-ui/pull/6#discussion_r1990066580 every single time you don't remove debugging code or comments from your PRs.

I need you to look through your PRs and double check them before you ask for a review.

I will review code correctness and style. I will not review unfinished code.